src/machine_learning/loss_function/negative_log_likelihood.rs

```diff
@@ -0,0 +1,77 @@
+// Negative Log Likelihood Loss Function
+//
+// The `neg_log_likelihood` function calculates the Negative Log Likelyhood loss,
+// which is a loss function used for classification problems in machine learning.
+//
+// ## Formula
+//
+// For a pair of actual and predicted values, represented as vectors `y_true` and
+// `y_pred`, the Negative Log Likelihood loss is calculated as:
+//
+// - loss = `-y_true * log(y_pred) - (1 - y_true) * log(1 - y_pred)`.
+//
+// It returns the average loss by dividing the `total_loss` by total no. of
+// elements.
+
+pub fn neg_log_likelihood(y_true: &[f64], y_pred: &[f64]) -> Option<f64> {
+    // Checks if the length of the actual and predicted values are equal
+    if y_true.len() != y_pred.len() || y_pred.is_empty() || y_true.is_empty() {
```

Comment:
What do you think about simplifying this condition?

```suggestion
    if y_true.len() != y_pred.len() || y_pred.is_empty() {
```
If I am not wrong, it is equivalent.

Reply:
@jkauerl: and here.

Reply:
I was thinking of leaving the condition because maybe some could input vectors with different sizes.

Reply:
> I was thinking of leaving the condition because maybe some could input vectors with different sizes.

They cannot be, because of `y_true.len() != y_pred.len()`.

Reply:
Oh right, just updated it.

Reply:
Although, I think is important to include the condition because there can be 2 errors associated. One related to an empty vector, and another with 2 vectors with different sizes. So then it would be neccessary to include both conditions?